### PR TITLE
[FIX] mail: do not return members of channels in channel_info

### PR DESCRIPTION
**Pull request by seb-odoo**
This PR has 3 parts:

The members fix itself, which is necessary for performance reasons especially
at init messaging, as per described in the task.

Adding the partner name in the typing payload, because due to not knowing all
members anymore, it is necessary to properly display who is typing.

Restrict restoring scroll position only if the corresponding scroll height is
available, to make the scroll feature and particularly the corresponding tests
more robust.

task-2360020

### CODE REVIEW

File: addons/mail/static/src/models/messaging_initializer/messaging_initializer.js [hunk 1]
```diff
@@ -153,9 +153,17 @@ function factory(dependencies) {
...
                 await this.async(() => new Promise(resolve => setTimeout(resolve)));
-                const channel = this.env.models['mail.thread'].insert(
-                    this.env.models['mail.thread'].convertData(channelData)
-                );
+                const convertedData = this.env.models['mail.thread'].convertData(channelData);
+                if (!convertedData.members) {
+                    // channel_info does not return all members of channel for
+                    // performance reasons, but code is expecting to know at
+                    // least if the current partner is member of it.
+                    // (eg. to know when to display "invited" notification)
```
Comment (alexkuhn) [inline addons/mail/static/src/models/messaging_initializer/messaging_initializer.js:161]: `eg.` => `e.g.`

**Review by alexkuhn — COMMENTED** [review]
Did you check some other features weren't broken, like partner member mention (especially without `hr` module) and typing notification on multi-user channels?

**Comment by seb-odoo** [comment]
Typing is "working" but we don't know the name of the person typing if it wasn't fetched from another means. I guess typing notif should send the name along with the id to fix the problem.

**Comment by seb-odoo** [comment]
@alexkuhn runbot finally green again, you should check this in priority (need it for prod in 1 week)

**Comment by alexkuhn** [comment]
How is the scroll height part of PR related to removal of channel members knowledge from `channel_info()`?
Or is it just something you added to make scroll adjustments more robust?

This looks alright, make sense that scrolltop adjustment should be applied with constant scrollheight. Too bad scroll adjustment cannot just happen after layout is fully rendered, but that's how it is...

I haven't checked it in detail, but I wouldn't be surprised having bugs later on if we don't nicer design of adjusting scrolls. This only works with the assumption that scrollheight never changes at all, which is fine for now.. but likely not the case when a rendering changes it. This is out-of-scope of PR, and we are fully aware scroll adjustment needs more improvements, like having this concern at a broader level (even up to models?). Still a long way to go before scroll adjustment is a problem no more.

**Comment by seb-odoo** [comment]
Scroll change is just to make runbot green by making these tests (and the actual feature scroll) more robust.

It's otherwise unrelated to members (now that I think about it I should have made a prep commit). I just could not get the members change to work without it

**Review by alexkuhn — APPROVED** [review]
@robodoo r+

**Pull request closed without merging**